api/src/main/java/org/pmiops/workbench/api/OfflineRuntimeController.java

```diff
@@ -357,14 +352,20 @@ private boolean notifyForUnusedDisk(LeonardoListPersistentDiskResponse disk, int
     return true;
   }
 
-  private boolean isDiskAttached(
-      LeonardoListPersistentDiskResponse diskResponse, String googleProject) throws ApiException {
+  private boolean isDiskAttached(ListPersistentDiskResponse diskResponse, String googleProject)
+      throws ApiException {
     final String diskName = diskResponse.getName();
 
     if (leonardoMapper.toApiListDisksResponse(diskResponse).getIsGceRuntime()) {
       return runtimesApiProvider.get().listRuntimesByProject(googleProject, null, false).stream()
-          .flatMap(runtime -> Stream.ofNullable(runtime.getDiskConfig()))
-          .anyMatch(diskConfig -> diskName.equals(diskConfig.getName()));
+          .flatMap(
+              runtime ->
+                  Stream.ofNullable(
+                      runtime
+                          .getRuntimeConfig()
+                          .getGceWithPdConfigInResponse()
+                          .getPersistentDiskId()))
+          .anyMatch(Objects::nonNull);
```

Comment:
we need the disk name to match like it did previously.  The reason is that we used to have a bug where we created multiple PDs for users, and some of these PD still exists. 
 So: it's possible that an *arbitrary* PD is not the *correct* PD to return here.

Reply:
see https://github.com/all-of-us/workbench/pull/7869#discussion_r1319098770